### PR TITLE
json rpc update

### PR DESCRIPTION
**Pull request by xrdavies**
* add method xdag_get_transactions
* fix memory leak
* use thread for rpc call
* refactory rpc_service, add rpc_procedure for easy usage

update based on #227

### CODE REVIEW

File: client/Makefile [hunk 1]
```diff
@@ -7,17 +7,17 @@ dfslib			= $(dfstools)/dfslib
...
 utils    		= ./utils
-moving_statistics 	= ./moving_statistics
+moving_statistics 	= ./utils/moving_statistics
 secp256k1               = ../secp256k1
 OS	:= $(shell uname)
 
 ifeq ($(OS), Darwin)
 	asm_src = 						\
-		sha256-mb-x86_64-mac.s				\
+		sha256-mb-x86_64-mac.s		\
```
Comment (jonano614) [inline client/Makefile]: Not an issue, but you broke alignment for a lot of \

Comment (xrdavies) [reply client/Makefile]: Quite interesting, I just aligned them with Sublime Text.

Comment (xrdavies) [reply client/Makefile]: image

Please see the alignment with vi

Comment (jonano614) [reply client/Makefile]: Evident example of different size of tabs in different systems

**Comment by jonano614** [comment]
Thank you for your great efforts!

**Pull request merged**